### DISCUSSION TITLE
error: no arguments passed inside the base constructor

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [Cyfrin/foundry-full-course-cu#2308] [Q&A] [answered]
Reference: `DecentralizedStableCoin.sol`

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {ERC20Burnable, ERC20} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";

/*
* @title DecentralizedStableCoin
* @author Sarthak Shah
* Collateral: Exogenous (ETH & BTC)
* Minting: Algorithmic
* Relative Stability: Pegged to USD
*
* This is the contract meant to be governed by DSCEngine. This contract is just the ERC20 implementation of our stablecoin system.
*
*/
contract DecentralizedStableCoin is ERC20Burnable, Ownable {
    error DecentralizedStableCoin__MustBeMoreThanZero();
    error DecentralizedStableCoin__BurnAmountExceedsBalance();
    error DecentralizedStableCoin__NotZeroAddress();

    constructor() ERC20("DecentralizedStableCoin", "DSC") {}

    function burn(uint256 _amount) public override onlyOwner{
        uint256 balance = balanceOf(msg.sender);
        if(_amount <= 0) {
            revert DecentralizedStableCoin__MustBeMoreThanZero();
        }
        if(balance < _amount){
            revert DecentralizedStableCoin__BurnAmountExceedsBalance();
        }
        super.burn(_amount);
    }

    function mint(address _to, uint256 _amount) external onlyOwner returns(bool){
        if(_to == address(0)){
            revert DecentralizedStableCoin__NotZeroAddress();
        }
        if(_amount <= 0){
            revert DecentralizedStableCoin__MustBeMoreThanZero();
        }
        _mint(_to, _amount);
        return true;
    }
}
```

While running `forge build` for the exact same code written by Patrick, he was able to run the program but I ran into this error:

```
Error (3415): No arguments passed to the base constructor. Specify the arguments or mark "DecentralizedStableCoin" as abstract.
  --> src/DecentralizedStableCoin.sol:18:1:
   |
18 | contract DecentralizedStableCoin is ERC20Burnable, Ownable {
   | ^ (Relevant source part starts here and spans across multiple lines).
Note: Base constructor parameters:
```

Can anyone tell me how to fix this?
Thanks

**Comment by Not-Sarthak** [comment, marked as answer]
After researching a bit, I found out that I had to pass address inside the base constructor as:
```
    constructor() ERC20("DecentralizedStableCoin", "DSC") Ownable(0x2e509A864c6376107155B0Bfb70f91FB370D876E) { }
```

However, when i copy paste Patrick's code without address as parameter, it still doesn't give me an error. Why?

**Reply by EngrPips** [reply]
Hello @Not-Sarthak, Patrick used a version of the `Ownable` contract where you didn't need to pass in any argument to the constructor of the `Ownable` contract, but if you import the latest version of the `Ownable` contract, it now requires that you pass in an argument in the constructor. The `chronological update` section treated this.

**Reply by Not-Sarthak** [reply]
But with the same version being installed, if I copy paste Patrick's code. It builds without any errors unlike in my case.

Let me know if I am missing something here? 😅

**Reply by EngrPips** [reply]
Interesting. You might want to find out why that is. Trace the path from which Patrick's code imported from and see what the implementation is.

**Reply by Not-Sarthak** [reply]
Tried doing that.
1. Tracing the path of my code [from imports]
2. Tracing the path of Patick's code [from his imports]

Both lead to the same path 😆 Now it's getting funny lol

**Reply by EngrPips** [reply]
Interesting, what is the implementation in that path? does it take a constructor argument or not?

**Reply by Not-Sarthak** [reply]
It does take the constructor argument but works even if not passed much similar to as if it is an optional in TypeScript 🤔 which obviously isn't possible in Solidity. Correct me if i am wrong here 😅

**Reply by EngrPips** [reply]
I am not sure how it takes a constructor argument, and it will still work even if you didn't pass the necessary constructor argument during deployment. You should investigate that further.

**Reply by Not-Sarthak** [reply]
On it, thank you!
